docs/zynq-7000-getting-started-guide.md

```diff
@@ -0,0 +1,183 @@
+
+This demonstration shows how to compile and run the Co-Simulation demo of Buildroot in QEMU with a simulated device in SystemC.  **Buildroot was selected for this demo, but other Linux Kernel's can be built and deployed in QEMU and used as well.** This configuration is tested working for Ubuntu 18.0.4 and assumes that a `cosim` directory is created in your home directory. This walkthrough also assumes that the device being emulated by QEMU is the Xilinx Zynq-7000 SoC.  This SoC seemed like a good candidate but the concept can apply to any QEMU machine which plugs in a compatible remoteport bus interface.
+
+## Dependencies
+
+Below are the dependencies needed to compile all the libraries in this demo:
+
+```bash
+sudo apt update
+sudo apt install cmake gmake gcc qemu-kvm qemu-system qemu-user-static verilator
+```
+
+## Setup and Compilation
+
+Run these commands to clone and build the necessary repos (`~/cosim` assumed as the base directory).
+
+### Create the base directory
+
+```bash
+mkdir ~/cosim
+```
+
+### SystemC Setup
+
+```bash
+# Build and install the binaries
+cd ~/cosim
+SYSC_VERSION=systemc-2.3.2
+wget https://www.accellera.org/images/downloads/standards/systemc/systemc-2.3.2.tar.gz
+tar xf systemc.2.3.2.tar.gz
+cd systemc-2.3.2/
+CXXFLAGS=-std=c++11 ./configure ; make
+sudo make install
+```
+
+### QEMU Setup
+
+```bash
+cd ~/cosim
+git clone https://github.com/Xilinx/qemu.git
+cd qemu
+git checkout 7e3e3ae09af8abe70383a5
+git submodule update --init dtc
+
+# Configure and build
+./configure --target-list="arm-softmmu,aarch64-softmmu,microblazeel-softmmu" --enable-fdt --disable-kvm --disable-xen
+make -j$((`nproc`+1))
+sudo make install
+export SYSTEM=/cosim/systemc-2.3.2/
+```
+
+### Demo Setup
+
+```bash
+# Clone the repo and the submodules
+cd ~/cosim
+git clone https://github.com/Xilinx/systemctlm-cosim-demo.git
+cd systemctlm-cosim-demo
+sed -i 's~../libsystemctlm-soc.git~https://github.com/Xilinx/libsystemctlm-soc.git~g' .gitmodules
+git submodule sync --recursive && git submodule update --init libsystemctlm-soc
+
+# Create the Makefile configeration
+cat << EOF | tee .config.mk
+CXXFLAGS=-std=c++11
+SYSTEMC=${HOME/cosim}
+HAVE_VERILOG=n
+HAVE_VERILOG_VERILATOR=n
+HAVE_VERILOG_VCS=n
+VM_TRACE=1
+EOF*
+
+# Build the demo
+make zynq_demo
+```
+
+### Buildroot Setup
+
+```bash
+cd ~/cosim
+git clone https://github.com/buildroot/buildroot.git
+cd buildroot
+git checkout 36edacce9c2c3b90f9bb11
+mkdir handles
+```
+
+For this walkthough we will be configuring the demo to emulate the Zynq-7000 SoC. 
+
+```bash
+# Pull the .dtsi files for the Zynq-7000
+wget https://raw.githubusercontent.com/Xilinx/qemu-devicetrees/master/zynq-pl-remoteport.dtsi
+wget https://raw.githubusercontent.com/Xilinx/linux-xlnx/master/arch/arm/boot/dts/zynq-7000.dtsi
+wget https://raw.githubusercontent.com/Xilinx/device-tree-xlnx/master/device_tree/data/kernel_dtsi/2017.3/BOARD/zc702.dtsi
```

Comment:
Change above wget to below (for fetching the zynq-zc702.dts and matching the name in BR2_LINUX_KERNEL_CUSTOM_DTS_PATH):
```
# kernel 5.4.75 zynq-zc702.dts
wget https://raw.githubusercontent.com/torvalds/linux/6e97ed6efa701db070da0054b055c085895aba86/arch/arm/boot/dts/zynq-zc702.dts
```
Also after downloading the dts/dtsi files we need to do below for including the remote port setups:
```
echo "#include \"zynq-pl-remoteport.dtsi\"" >> zynq-zc702.dts
```